### PR TITLE
Refactor: Integrate EthCallBundle Module into Eth Module

### PR DESCRIPTION
**Pull request by nnlgsakib**
This PR refactors the RethRpcModule enum by removing the EthCallBundle variant and integrating its functionality directly into the Eth module.

Key Changes:
- Removed EthCallBundle Variant: The EthCallBundle variant has been deprecated and its features are now fully encompassed within the Eth module.
- Updated RpcModuleSelection: Adjusted the module selection logic to reflect the removal of EthCallBundle.
- Documentation Update: Modified code comments and documentation to align with the new module structure.
This refactor simplifies the RPC module structure by consolidating related functionalities under the Eth module, improving code maintainability and clarity.

### CODE REVIEW

**Comment by emhane** [comment]
> * Removed EthCallBundle Variant: The EthCallBundle variant has been deprecated and its features are now fully encompassed within the Eth module.
do you have a link to specs showing this is deprecated pls?

**Comment by emhane** [comment]
ah you're referring to this issue https://github.com/paradigmxyz/reth/issues/10483 ? I think we have competing pr https://github.com/paradigmxyz/reth/pull/10486#event-13996037941, right?

**Comment by nnlgsakib** [comment]
> ah you're referring to this issue #10483 ? I think we have competing pr #10486, right?

yah

**Comment by tcoratger** [comment]
> ah you're referring to this issue #10483 ? I think we have competing pr #10486, right?

@emhane Yeah seems so, but feel free to take the one you prefer as soon as the issue is resolved :)

File: crates/rpc/rpc-server-types/src/module.rs [hunk 1]
```diff
@@ -251,87 +252,27 @@ pub enum RethRpcModule {
...
     Rpc,
-    /// `reth_` module
-    Reth,
-    /// `ots_` module
-    Ots,
-    /// For single non-standard `eth_` namespace call `eth_callBundle`
-    ///
-    /// This is separate from [`RethRpcModule::Eth`] because it is a non standardized call that
-    /// should be opt-in.
-    EthCallBundle,
```
Comment (tcoratger) [inline crates/rpc/rpc-server-types/src/module.rs:262]: Why do we need to remove all of these and not only `EthCallBundle`?

**Comment by nnlgsakib** [comment]
👍

**Pull request closed without merging**